src/nag-pack.ts

```diff
@@ -229,18 +237,22 @@ export abstract class NagPack implements IAspect {
    * @param findingId The id of the finding.
    * @param info Why the rule was triggered.
    * @param explanation Why the rule exists.
+   * @param metadata custom metadata for the rule.
    * @returns The formatted message string.
    */
   protected createMessage(
     ruleId: string,
     findingId: string,
     info: string,
-    explanation: string
+    explanation: string,
+    metadata: string = ''
   ): string {
     let message = findingId
       ? `${ruleId}[${findingId}]: ${info}`
       : `${ruleId}: ${info}`;
-    return this.verbose ? `${message} ${explanation}\n` : `${message}\n`;
```

Comment:
This seems like it's an extension of the explanation field.  Additionally If it's arbitrary 'internal use'  metadata then it doesn't make sense to present it the user